packages/teleterm/src/ui/services/usageEvent/usageEventService.ts

```diff
@@ -0,0 +1,166 @@
+
+export class UsageEventService {
+  constructor(
+    private tshClient: TshClient,
+    private configService: ConfigService,
+    private findCluster: (clusterUri: ClusterUri) => Cluster,
+    private runtimeSettings: RuntimeSettings
+  ) {}
+
+  captureLogin(uri: ClusterOrResourceUri): Promise<void> {
+    const clusterParams = this.getClusterProperties(uri);
+    if (!clusterParams) {
+      return;
+    }
+    return this.reportUsageEvent({
+      loginEvent: {
+        clusterProperties: clusterParams,
+        arch: this.runtimeSettings.arch,
+        os: this.runtimeSettings.platform,
+        osVersion: this.runtimeSettings.osVersion,
+        connectVersion: this.runtimeSettings.connectVersion,
+      },
+    });
+  }
+
+  captureProtocolRun(
+    uri: ClusterOrResourceUri,
+    protocol: 'ssh' | 'kube' | 'db'
+  ): Promise<void> {
+    const clusterProperties = this.getClusterProperties(uri);
+    if (!clusterProperties) {
+      return;
+    }
+    return this.reportUsageEvent({
+      protocolRunEvent: {
+        clusterProperties,
+        protocol,
+      },
+    });
+  }
+
+  captureAccessRequestCreate(
+    uri: ClusterOrResourceUri,
+    kind: 'role' | 'resource'
+  ): Promise<void> {
+    const clusterProperties = this.getClusterProperties(uri);
+    if (!clusterProperties) {
+      return;
+    }
+    return this.reportUsageEvent({
+      accessRequestCreateEvent: {
+        clusterProperties,
+        kind,
+      },
+    });
+  }
+
+  captureAccessRequestReview(uri: ClusterOrResourceUri): Promise<void> {
+    const clusterProperties = this.getClusterProperties(uri);
+    if (!clusterProperties) {
+      return;
+    }
+    return this.reportUsageEvent({
+      accessRequestReviewEvent: {
+        clusterProperties,
+      },
+    });
+  }
+
+  captureAccessRequestAssumeRole(uri: ClusterOrResourceUri): Promise<void> {
+    const clusterProperties = this.getClusterProperties(uri);
+    if (!clusterProperties) {
+      return;
+    }
+    return this.reportUsageEvent({
+      accessRequestAssumeRoleEvent: {
+        clusterProperties,
+      },
+    });
+  }
+
+  captureFileTransferRun(
+    uri: ClusterOrResourceUri,
+    direction: 'download' | 'upload'
+  ): Promise<void> {
+    const clusterProperties = this.getClusterProperties(uri);
+    if (!clusterProperties) {
+      return;
+    }
+    return this.reportUsageEvent({
+      fileTransferRunEvent: {
+        clusterProperties,
+        direction,
+      },
+    });
+  }
+
+  captureUserJobRoleUpdate(jobRole: string): Promise<void> {
+    return this.reportUsageEvent({
+      userJobRoleUpdateEvent: {
+        jobRole,
+      },
+    });
+  }
+
+  private reportUsageEvent(event: ReportEventRequest['event']): Promise<void> {
+    const isCollectingUsageMetricsEnabled = this.configService.get(
+      'usageMetrics.enabled'
+    ).value;
+
+    if (!isCollectingUsageMetricsEnabled) {
+      return;
+    }
+    if (this.runtimeSettings.dev) {
+      return;
+    }
```

Comment:
> We could even change the default version string for the dev version from 1.0.2 to 1.0.0-dev so that someone can filter those out if they want to. 

Brilliant idea! I had some doubts about being dependant on `runtimeSettings.dev` (like you have to disable this check while you develop this feature), but I didn't have any better solution. I'm happy to change it.